### ISSUE TITLE
Script doesn`t work properly on three aspects

### ISSUE THREAD

**Issue by eirin-yukari** [Fate-Grand-Automata/FGA#1277]
### Preparation

- [X] I tested the [latest release](https://github.com/Fate-Grand-Automata/FGA/releases/tag/canary)
- [X] I looked at other issues (even the [closed ones](https://github.com/Fate-Grand-Automata/FGA/issues?q=is%3Aissue+is%3Aclosed))
- [X] I read the [Troubleshooting Guide](https://github.com/MathewSachin/Fate-Grand-Automata/wiki/Troubleshooting)

### FGO server

CN

### FGA build number

1745

### Describe the bug

FGA cannot select typical mission or supportive servant, it needs manual pointing, but it can still make effort on battles. Besides, despite of pointing skills to various servants, FGA always points to the middle servant.
![1](https://user-images.githubusercontent.com/70759792/182096345-53ecbb37-8c0c-42bb-8d69-5f34303a77df.jpg)
i am sorry that the recording function of FGA didn`t work on some reasons, so i used the recording function of Andriod. Hope it works 

### Video

https://user-images.githubusercontent.com/70759792/182098473-b13378ca-d81c-4058-b20b-1a536f67ca16.mp4

### Device model

Huawei mate 40 pro

### Android version

10

### Screen size

2772*1344

**Comment by reconman** [comment]
Is your Skill Confirmation setting turned off in FGA?

**Comment by eirin-yukari** [comment]
> Is your Skill Confirmation setting turned off in FGA?

Yes,because FGA will point skills to the third servant if i open it

**Comment by reconman** [comment]
The wrong skill is used because FGA thinks that both Skadis are dead.

I think the black bars at the top and bottom are the reason for the bug.

If you post a screenshot, I can measure the black bars for you so you can use Custom Game Area mode.

**Comment by eirin-yukari** [comment]
> 

![2](https://user-images.githubusercontent.com/70759792/182282975-8b01f289-0c40-4ab5-b38c-b408868e459b.jpg)

**Comment by eirin-yukari** [comment]
> The wrong skill is used because FGA thinks that both Skadis are dead.
> 
> I think the black bars at the top and bottom are the reason for the bug. Huawei adds the black bars because of the curved edges.
> 
> If you post a screenshot, I can measure the black bars for you so you can use Custom Game Area mode.
> 
> Or you can try to remove them in Settings > Display & brightness > More display settings > Edges

Well, thanks to your advice, I found that the FGA doesn’t match the curved surface of Huawei mate 40 pro, so I turned off the expansion display of left and right border of FGA on settings which it turned on automatically, now it can point skills to correct servants and pick up correct mission. However, it still can’t select supportive servant.

**Comment by reconman** [comment]
I think you have to do the same for FGO.

**Comment by eirin-yukari** [comment]
> I think you have to do the same for FGO.

Unfortunately, the same operation seems ineffective. I can’t descry the black bar after I turn off the setting of FGO. Besides, like other issues before, FGA can’t recognize the card of Buster type.

**Comment by reconman** [comment]
Can you post a screenshot of FGO?

**Comment by eirin-yukari** [comment]
> Can you post a screenshot of FGO?

![3](https://user-images.githubusercontent.com/70759792/182562455-daf19a57-15a3-48e4-af3b-1425c485519b.jpg)

**Comment by reconman** [comment]
Can you try Custom Game Area with top=28 and bottom=28?

**Comment by eirin-yukari** [comment]
> Can you try Custom Game Area with top=28 and bottom=28?

Err, it’s quite upset that it causes new problems of recognize on that setting. So I recorded a video.

https://user-images.githubusercontent.com/70759792/183024528-7fd9399d-88e0-4caf-995e-447ad5263c1f.mp4

**Comment by lianli2352** [comment]
我的小米也有这个问题，日服没做全面屏优化导致两边黑边有问题，直接卡在了选择职阶那一步，老哥你的问题解决了没😭

**Comment by reconman** [comment]
@eirin-yukari Try again with top and bottom = 13.

**Comment by eirin-yukari** [comment]
> @eirin-yukari Try again with top and bottom = 13.

FGA can select skills in missions now with that setting, but it still can’t select typical mission and supportive servant. Besides, it can’t recognize any color of cards in missions now.

**Comment by reconman** [comment]
You'll have to test different curved screen app settings and offsets yourself now. Some combination will work.

I can't help you any further.

**Comment by eirin-yukari** [comment]
> You'll have to test different curved screen app settings and offsets yourself now. Some combination will work.
> 
> I can't help you any further.

Ok, I am truly thankful you can help me. I will report on this issue if I find the solution.
